File: 0228_Summary_Ranges.go
import "fmt"

func summaryRanges(nums []int) []string {
    var i,j int
    fout := make([]string,0)
    if len(nums) == 0{
        return fout
    }
    for j < len(nums)-1{
        if nums[j+1] != nums[j]+1{
            if i != j{
                fout = append(fout,fmt.Sprintf("%d->%d",nums[i],nums[j]))
            } else{
                fout = append(fout,fmt.Sprintf("%d",nums[j]))
            }
            i = j+1
        }
        j++
    }
    if i < j{
        fout = append(fout,fmt.Sprintf("%d->%d",nums[i],nums[len(nums)-1]))
    }else{
        fout = append(fout,fmt.Sprintf("%d",nums[len(nums)-1]))
    }
    return fout
}

File: 0228_Summary_Ranges.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        fout = []
        if nums == []:
            return fout
        i,j = 0,0
        while j < len(nums)-1:
            if nums[j+1] != nums[j] + 1:
                if i == j:
                    fout.append(str(nums[i]))
                else:
                    fout.append("{:d}->{:d}".format(nums[i],nums[j]))
                i = j+1
            j += 1
        #Final entry
        if i == j:
            fout.append(str(nums[-1]))
        else:
            fout.append("{:d}->{:d}".format(nums[i],nums[-1]))
        return fout

